tests/dcgm_test.go

```diff
@@ -42,6 +43,34 @@ func BenchmarkDeviceCount1(b *testing.B) {
 	dcgm.Shutdown()
 }
 
+func TestCpuQuery(t *testing.T) {
+	os.Setenv("DCGM_SKIP_SYSMON_HARDWARE_CHECK", "1")
+	cleanup, err := dcgm.Init(dcgm.Embedded)
```

Comment:
Can we use mocks instead of running on actual hardware? Also, we can separate unit tests from integration tests, for example as described here: https://konradreiche.com/blog/how-to-separate-integration-tests-in-go

Reply:
While I agree that we should use mocks in the near future, all the tests in this file (actually all go-dcgm tests) require actual hardware so I would prefer stick with that pattern for now and add mocks in a future commit.